### PR TITLE
feat: add multiple new certificate events test

### PR DESCRIPTION
**Pull request by atanmarko**
# Description

Add test with multiple send token events

Fixes TP-691

## PR Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

File: crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs
Comment (dvdplm) [inline crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs:128]:
```suggestion
    let subnet_startup_delay = if subnet_type == STANDALONE_SUBNET_WITH_LONG_BLOCKS {
```

File: crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs
Comment (dvdplm) [inline crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs:984]:
```suggestion
// Test multiple send token events in a block
// Test is slow, block time is 12 seconds
```

(While it's great to have comments describing a test, we do not want `rustdoc` to generate documentation for them)

File: crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs
Comment (dvdplm) [inline crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs:988]: What do you think about adding `#[timeout]` here, i.e. https://github.com/la10736/rstest#test-timeout

Comment (atanmarko) [reply crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs:989]: It has to be some very big failsafe timeout, because it is hard to assume how long the test will last - it has many steps with long per block time

**Comment by atanmarko** [comment]
Note: Polygon Edge related [PR](https://github.com/topos-protocol/polygon-edge/pull/29) must be merged for this test to work

File: crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs
Comment (atanmarko) [reply crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs:993]: Right, I already put 10 minutes

**Comment by dvdplm** [comment]
> Note: Polygon Edge related [PR](https://github.com/topos-protocol/polygon-edge/pull/29) must be merged for this test to work

Oh. Hmm. How can I run this PR with that PR applied?

**Comment by dvdplm** [comment]
@atanmarko Is this the reason CI is failing?

**Comment by atanmarko** [comment]
> @atanmarko Is this the reason CI is failing?

I had some additional fixes

**Comment by atanmarko** [comment]
> > Note: Polygon Edge related [PR](https://github.com/topos-protocol/polygon-edge/pull/29) must be merged for this test to work
> 
> Oh. Hmm. How can I run this PR with that PR applied?

You can checkout `polygon-edge` PR referenced, build local docker image:
```
docker build -t ghcr.io/topos-protocol/polygon-edge:develop --build-arg GITHUB_TOKEN=${GITHUB_TOKEN} -f docker/local-topos/Dockerfile .
```
Then run topos tests:
```
cd crates/topos-sequencer-subnet-runtime
cargo test 
```

**Comment by atanmarko** [comment]
Put in a draft mode until Polygon Edge fix is merged

File: crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs
Comment (Freyskeyd) [inline crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs:236]:
```suggestion
        wallet.with_chain_id(chain_id.as_u64()),
```

File: crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs
Comment (Freyskeyd) [inline crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs:242]: This method is really similar to the one above, maybe we can merge or share some part of it

**Review by Freyskeyd — APPROVED** [review]

File: crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs
Comment (dvdplm) [inline crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs:1273]:
```suggestion
    info!("Transferring tokens to {} accounts", test_accounts.len());
```

File: crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs
Comment (dvdplm) [inline crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs:1329]: The comment seems redundant here.

File: crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs
Comment (dvdplm) [inline crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs:1465]:
```suggestion
    // Set big timeout to prevent flaky failures. Instead fail/panic early in the test to indicate actual error
```

**Review by dvdplm — APPROVED** [review]
lgtm modulo @Freyskeyd's comments. Left a few nitpicks to fix too.

File: crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs
Comment (atanmarko) [reply crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs:242]: Ok, I'll work a lot more on the sequencer tests in the foreseeable future so I'll optimize it

**Pull request merged**
